app/Libraries/OsuAuthorize.php

```diff
@@ -1561,25 +1562,34 @@ public function checkForumTopicPollEdit(?User $user, Topic $topic): string
     }
 
     /**
-     * @param User|null $user
-     * @param Topic $topic
-     * @return string
+     * @throws AuthorizationCheckException
+     */
+    public function checkForumTopicPollOptionShowResult(?User $user, PollOption $pollOption): string
+    {
+        return $this->checkForumTopicPollShowResults($user, $pollOption->topic);
+    }
+
+    /**
+     * @throws AuthorizationCheckException
      */
     public function checkForumTopicPollShowResults(?User $user, Topic $topic): string
     {
-        if (!$topic->poll_hide_results) {
+        if (!$topic->poll_hide_results || ($topic->pollEnd()?->isPast() ?? true)) {
             return 'ok';
         }
 
-        if ($this->doCheckUser($user, 'ForumModerate', $topic->forum)->can()) {
-            return 'ok';
+        $this->ensureLoggedIn($user);
+
+        $isNotOAuthPermission = $this->doCheckUser($user, 'IsNotOAuth');
+        if (!$isNotOAuthPermission->can()) {
+            return $isNotOAuthPermission->rawMessage();
         }
 
-        if ($topic->pollEnd() === null || $topic->pollEnd()->isPast()) {
+        if ($this->doCheckUser($user, 'ForumModerate', $topic->forum)->can()) {
             return 'ok';
         }
 
-        if ($user !== null && $topic->posts()->withTrashed()->first()->poster_id === $user->user_id) {
+        if ($topic->posts()->withTrashed()->first()->poster_id === $user->getKey()) {
```

Comment:
This part isn't reachable using oauth but this doesn't use preloaded `firstPost` (and maybe just use `topic_poster` instead)